pkg/semtech-loramac/Makefile.dep

```diff
@@ -6,6 +6,9 @@ USEMODULE += semtech_loramac_mac_region
 USEMODULE += semtech_loramac_crypto
 USEMODULE += semtech_loramac_arch
 
+USEMODULE += ztimer_msec
+USEMODULE += ztimer_periph_rtt
```

Comment:
```suggestion
```suggestion
# If RTT feature is available use the RTT backend of ztimer
FEATURES_OPTIONAL += periph_rtt
ifneq (,$(filter periph_rtt,$(FEATURES_USED)))
  USEMODULE += ztimer_periph_rtt
endif
```
```

Reply:
I prefer keeping the actual state for the loramac pkg: it's important to ensure that the rtt is used because of the duty-cycle timer. In a deep sleep mode with RAM retention, a regular timer will be stopped but not a rtt based timer.

Reply:
But why enforce it? Low Power modes are not enabled by default, only `examples/lorawan` does that.

Reply:
ztimer is an internal implementation detail of the package adaption to RIOT. There's no need to enforce developers to add this dependency in their lorawan applications.

Reply:
I think is a question of keeping flexibility or having things that work out of the box. I think you want the later, that is OK for me since you are maintaining this. Can you then add some documentation to the pkg on the implementation status, I think it would be good to say why `rtt` is required or when it is required. And in the same sense, what is needed for the pkg to work, what is needed for the pkg to work with sleep modes, etc... For me this is not very clear without having a knowledge of the code/implementation internals.